linc_convert/utils/nifti_header.py

```diff
@@ -0,0 +1,243 @@
+"""Utilities for loading, creating, and manipulating NIfTI headers."""
+import io
+import warnings
+from pathlib import Path
+from typing import Optional, Tuple
+
+import nibabel as nib
+import numpy as np
+from niizarr import default_nifti_header
+from niizarr._header import bin2nii, get_nibabel_klass
+
+from linc_convert.utils.io.zarr import open_array
+from linc_convert.utils.io.zarr.abc import NiftiHeader
+from linc_convert.utils.orientation import center_affine, orientation_to_affine
+from linc_convert.utils.unit import to_nifti_unit
+from linc_convert.utils.zarr_config import NiftiConfig
+
+
+def _try_open_zarr_array(path: Path) -> "ZarrArray":  # noqa: F821
+    """
+    Attempt to open a Zarr array at `path`.
+
+    Returns the opened array on success, or raises the last exception.
+    """
+    last_err: Exception | None = None
+    for v in (2, 3):
+        try:
+            return open_array(path, mode="r", zarr_version=v)
+        except Exception as e:  # noqa: BLE001 - we want to keep original exception
+            # types
+            last_err = e
+    # If we get here, all attempts failed.
+    assert last_err is not None
+    raise last_err
+
+
+def load_nifti_header_from_file(path: str | Path) -> "NiftiHeader":
+    """
+    Load a NIfTI header from a file path.
+
+    Supports:
+      - Standard NIfTI files: `.nii`, `.nii.gz`
+      - Zarr containing Nifti Header Array: `.zarr`
+      - A Zarr Array containing a NIfTI header
+
+    Parameters
+    ----------
+    path : str | pathlib.Path
+        Path to a NIfTI file or Zarr directory.
+
+    Returns
+    -------
+    NiftiHeader
+        A nibabel NIfTI header object.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the path does not exist.
+    ValueError
+        If the path exists but is not a recognized NIfTI file or Zarr directory.
+    Exception
+        If Zarr opening fails for all supported versions.
+    """
+    p = Path(path)
+    if not p.exists():
+        raise FileNotFoundError(f"Path does not exist: {p}")
+
+    suffix = "".join(p.suffixes).lower()
+
+    # Standard NIfTI files
+    if suffix.endswith(".nii") or suffix.endswith(".nii.gz"):
+        img = nib.load(str(p))
+        return img.header.copy()
+
+    if suffix.endswith(".zarr") or p.is_dir():
+        zarr_target = p
+        candidate = p / "nifti"
+        if candidate.exists():
+            zarr_target = candidate
+        return load_nifti_header_from_zarray(zarr_target)
```

Comment:
If the `/nifti` directory doesn't exist, then should we throw an error since the Zarr doesn't follow the nifti-zarr spec and the nifti header won't be found?

Reply:
Or is this assuming that the argument passed in could be the `test.nii.zarr/nifti` directory?